bindings/node/lib/clientEncryption.js

```diff
@@ -693,6 +699,61 @@ module.exports = function (modules) {
     static get libmongocryptVersion() {
       return mc.MongoCrypt.libmongocryptVersion;
     }
+
+    /**
+     * A helper that perform explicit encryption of values and expressions.
+     * Explicitly encrypt a provided value. Note that either `options.keyId` or `options.keyAltName` must
+     * be specified. Specifying both `options.keyId` and `options.keyAltName` is considered an error.
+     *
+     * @internal
+     *
+     * @param {*} value The value that you wish to encrypt. Must be of a type that can be serialized into BSON
+     * @param {boolean} expressionMode - a boolean that indicates whether or not to encrypt the value as an expression
+     * @param {EncryptOptions} options
+     * @returns the raw result of the call to stateMachine.execute().  When expressionMode is set to true, the return
+     *          value will be a bson document.  When false, the value will be a BSON Binary.
+     */
+    async _encrypt(value, expressionMode, options) {
+      const bson = this._bson;
+      const valueBuffer = bson.serialize({ v: value });
+      const contextOptions = Object.assign({}, options, { expressionMode });
+      if (options.keyId) {
+        contextOptions.keyId = options.keyId.buffer;
+      }
+      if (options.keyAltName) {
+        const keyAltName = options.keyAltName;
+        if (options.keyId) {
+          throw new TypeError(`"options" cannot contain both "keyId" and "keyAltName"`);
+        }
+        const keyAltNameType = typeof keyAltName;
+        if (keyAltNameType !== 'string') {
+          throw new TypeError(
+            `"options.keyAltName" must be of type string, but was of type ${keyAltNameType}`
+          );
+        }
+
+        contextOptions.keyAltName = bson.serialize({ keyAltName });
+      }
+
+      const rangePreviewRegex = /^rangepreview$/i;
+      if (typeof options.algorithm === 'string' && rangePreviewRegex.test(options.algorithm)) {
```

Comment:
Absolutely non-blocking, but I don’t think you really need a regexp here:

```suggestion
      if (
        typeof options.algorithm === 'string' && 
        options.algorithm.toLowerCase() === 'rangepreview'
      ) {
```

Reply:
I'll remove this in favor of checking in C++